File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function removeDuplicates(&$nums) {
        $map = array();
        $n = sizeof($nums);
        $k = 0;
        for($i=0; $i<$n; ++$i){
            if(!isset($map[$nums[$i]])){
                $map[$nums[$i]] = 1;
                ++$k;
            }
            else $nums[$i] = '_';
        }
        sort($nums);
        return $k;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        HashMap<Integer, Boolean> map = new HashMap<>();
        int n = nums.length;
        int k = 0;
        for(int i=0; i<n; ++i){
            if(!map.containsKey(nums[i])){
                map.put(nums[i], true);
                ++k;
            }
            else nums[i] = 101;
        }
        Arrays.sort(nums);
        return k;
    }
}